std/datetime/mod.ts

```diff
@@ -150,3 +149,129 @@ export function toIMF(date: Date): string {
     months[date.getUTCMonth()]
   } ${y} ${h}:${min}:${s} GMT`;
 }
+
+/**
+ * Check given year is a leap year or not.
+ * @param year year in number or Date format
+ */
+export function isLeap(year: Date | number): boolean {
+  year = year instanceof Date ? year.getFullYear() : year;
+  return !(year % 400) || (year % 4 === 0 && year % 100 !== 0);
+}
+
+export type Unit =
+  | "miliseconds"
+  | "seconds"
+  | "minutes"
+  | "hours"
+  | "days"
+  | "weeks"
+  | "months"
+  | "quarters"
+  | "years";
+
+export type DifferenceFormat = {
+  miliseconds?: number;
+  seconds?: number;
+  minutes?: number;
+  hours?: number;
+  days?: number;
+  weeks?: number;
+  months?: number;
+  quarters?: number;
+  years?: number;
+};
+
+export type DifferenceOptions = {
+  units?: Unit[];
+};
+
+/**
+ * Calculate difference between two dates.
+ * @param from Year to calculate difference
+ * @param to Year to calculate difference with
+ * @param options Options for determining how to respond
+ *
+ * example :
+ *
+ * ```typescript
+ * datetime.difference(new Date("2020/1/1"),new Date("2020/2/2"),{ units : ["days","months"] })
+ * ```
+ */
+export function difference(
+  from: Date,
+  to: Date,
+  options?: DifferenceOptions
+): DifferenceFormat {
+  const uniqueUnits = options?.units
+    ? [...new Set(options?.units)]
+    : [
+        "miliseconds",
+        "seconds",
+        "minutes",
+        "hours",
+        "days",
+        "weeks",
+        "months",
+        "quarters",
+        "years",
+      ];
+
+  const bigger = from > to ? from : to;
+  const smaller = from > to ? to : from;
+  const differenceInMs = bigger.getTime() - smaller.getTime();
+
+  function calculateMonthDifference(biger: Date, smaller: Date): number {
+    const yearDiff = biger.getFullYear() - smaller.getFullYear();
+    const monthsDiff = biger.getMonth() - smaller.getMonth();
+    const calendarDiffrences = Math.abs(yearDiff * 12 + monthsDiff);
+    const compareResult = biger > smaller ? 1 : -1;
+    biger.setMonth(biger.getMonth() - compareResult * calendarDiffrences);
+    const isLastMonthNotFull =
+      biger > smaller ? 1 : -1 === -compareResult ? 1 : 0;
+    const months = compareResult * (calendarDiffrences - isLastMonthNotFull);
+    return months === 0 ? 0 : months;
+  }
+
+  const differences: DifferenceFormat = {};
+
+  for (let i = 0; i < uniqueUnits.length; i++) {
+    switch (uniqueUnits[i]) {
```

Comment:
since you don't really need the `i` variable, prefer a `for...of` loop here

```suggestion
  for (const uniqueUnit of uniqueUnits) {
    switch (uniqueUnit) {
```